### PR TITLE
EDSC-3358: Adding related collections to the collection details page

### PR DESCRIPTION
**Pull request by trevorlang**
# Overview

### What is the feature?

Adding related collections and metrics events to the collection details page

### What is the Solution?

Display top 3 related collections on the page

### What areas of the application does this impact?

Collection details page

# Testing

### Reproduction steps

- **Environment for testing:** ANY
- **Collection to test with:** ANY with related collections

1. View collection
2. Click related collection
3. Confirm granules page is displayed
4. Confirm metrics events fire

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: cypress/integration/paths/search/granules/granule_details/granule_details_spec.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import formattedGranuleMetadata from './__mocks__/formattedGranuleMetadata.json'
 
 describe('Path /search/granules/granule-details', () => {
   describe('When loading a granule', () => {
-    it('loads correctly', () => {
+    it.only('loads correctly', () => {
```
Comment (trevorlang) [inline cypress/integration/paths/search/granules/granule_details/granule_details_spec.js:14]: remove only

File: static/src/js/actions/focusedCollection.js [hunk 2]
```diff
@@ -111,6 +111,17 @@ export const getFocusedCollection = () => async (dispatch, getState) => {
         points
         polygons
         relatedUrls
+        relatedCollections {
```
Comment (trevorlang) [inline static/src/js/actions/focusedCollection.js:114]: limit

File: static/src/js/components/CollectionDetails/CollectionDetailsBody.js [hunk 3]
```diff
@@ -507,6 +518,61 @@ export const CollectionDetailsBody = ({
...
           }
+          {
+            !!(relatedCollectionsList.length && relatedCollectionsList.length > 0) && (
+              <div className="row collection-details-body__row collection-details-body__feature">
+                <div className="col col-12">
+                  <div className="collection-details-body__feature-heading">
+                    <h4 className="collection-details-body__feature-title">You might also be interested in...</h4>
+                    <ul className="collection-details-body__related-collections-list">
+                      {
+                        relatedCollectionsList.slice(0, 3).map((relatedCollection) => {
```
Comment (trevorlang) [inline static/src/js/components/CollectionDetails/CollectionDetailsBody.js:529]: Remove slice

File: cypress/integration/paths/search/granules/__mocks__/no_params/graphql.body.json [hunk 4]
```diff
@@ -56,6 +64,102 @@
       "nativeDataFormats": [],
       "points": null,
       "polygons": null,
+      "relatedCollections": {
```
Comment (trevorlang) [inline cypress/integration/paths/search/granules/__mocks__/no_params/graphql.body.json:67]: Fix this for limit

File: static/src/js/components/CollectionDetails/CollectionDetailsBody.js [hunk 5]
```diff
@@ -123,6 +129,7 @@ export const CollectionDetailsBody = ({
     hasAllMetadata,
     gibsLayers,
     nativeDataFormats,
+    relatedCollections = {},
```
Comment (abbottry) [inline static/src/js/components/CollectionDetails/CollectionDetailsBody.js:132]: Did you try a collection that has no related collections? I want to make sure its not returning null here which would ignore the default.

**Review by abbottry — APPROVED** [review]

**Pull request merged**